src/main/java/dev/sigstore/fulcio/client/SigningCertificate.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.fulcio.client;
+
+import com.google.api.client.util.PemReader;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import dev.sigstore.json.GsonSupplier;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.*;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.conscrypt.ct.DigitallySigned;
+import org.conscrypt.ct.SerializationException;
+import org.conscrypt.ct.SignedCertificateTimestamp;
+
+/** Response from Fulcio that includes a certPath and an SCT */
+public class SigningCertificate {
+
+  private final CertPath certPath;
+  @Nullable private final SignedCertificateTimestamp sct;
+
+  static SigningCertificate newSigningCertificate(String certs, @Nullable String sctHeader)
+      throws CertificateException, IOException, SerializationException {
+    CertPath certPath = decodeCerts(certs);
+    if (sctHeader != null) {
+      SignedCertificateTimestamp sct = decodeSCT(sctHeader);
+      return new SigningCertificate(certPath, sct);
+    }
+    return new SigningCertificate(certPath, null);
+  }
+
+  @VisibleForTesting
+  static CertPath decodeCerts(String content) throws CertificateException, IOException {
+    PemReader pemReader = new PemReader(new StringReader(content));
+    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+    ArrayList<X509Certificate> certList = new ArrayList<>();
+    while (true) {
+      PemReader.Section section = pemReader.readNextSection();
+      if (section == null) {
+        break;
+      }
+
+      byte[] certBytes = section.getBase64DecodedBytes();
+      certList.add((X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certBytes)));
+    }
+    if (certList.isEmpty()) {
+      throw new IOException("no valid PEM certificates were found in response from Fulcio");
```

Comment:
Is this maybe like an IllegalArgument or IllegalFormat exception? I have no idea, but IO feels a little weird since the IO has happened but the contents are not as expected.

Reply:
yeah it's def not an IOException, lemme fix that.